mesonpy/__init__.py

```diff
@@ -876,8 +876,12 @@ def build_commands(self, install_dir: Optional[pathlib.Path] = None) -> Sequence
     @functools.lru_cache(maxsize=None)
     def build(self) -> None:
         """Trigger the Meson build."""
-        for cmd in self.build_commands():
-            self._meson(*cmd[1:])
+        self._meson('compile', *self._meson_args['compile'],)
+        if self._meson_args['install-tags']:
+            install_tags = '--tags=' + ','.join(self._meson_args['install-tags'])
+            self._meson('install', '--destdir', os.fspath(self._install_dir), install_tags, *self._meson_args['install'],)
+        else:
+            self._meson('install', '--destdir', os.fspath(self._install_dir), *self._meson_args['install'],)
```

Comment:
I think this may have happened because of the merge, but this code now needs to go into `build_commands`.

Reply:
True, this is a merge error, my code was still based on a very old version of meson-python

Reply:
if install-tags is not separated from the "install" args, this whole block can be reverted to
`for cmd in self.build_commands():
    self._meson(*cmd[1:])`